mlflow/langchain/__init__.py

```diff
@@ -800,10 +801,23 @@ def load_model(model_uri, dst_path=None):
     return _load_model_from_local_fs(local_model_path)
 
 
-def _load_code_model(code_path: Optional[str] = None):
+@contextmanager
+def _config_path_context(code_path: Optional[str] = None):
     _set_config_path(code_path)
+    try:
+        yield
+    finally:
+        _set_config_path(None)
+
 
-    import chain  # noqa: F401
+def _load_code_model(code_path: Optional[str] = None):
+    with _config_path_context(code_path):
+        try:
+            import chain  # noqa: F401
+        except Exception as e:
```

Comment:
We should be catching an ImportError here, right?

Reply:
I initially thought that we should catch any error that comes from running the chain code. But I agree, here we should just catch the import error